### ISSUE TITLE
Filter Loop when using ? in alias

### ISSUE THREAD

**Issue by s3m1s0n1c** [pagermon/pagermon#278]
When having a ? in the alias name eg: dungog? this can cause a huge loop when doing a filter/search query on the alias name

when I do a search query on https://compagers.rfspager.com/?q=Dungog? it search's then flaps between that and https://compagers.rfspager.com/?q=Dungog?%3F

This causes MASSIVE load on the server and will end up killing the server or node app if using PM2 if left long enough

**Comment by davidmckenzie** [comment]
Welp, that's bad... tested and confirmed

**Comment by DanrwAU** [comment]
Probably a bug in the way I rewrote the search on v3 unless you tested in 2.3?

**Comment by davidmckenzie** [comment]
It's probably this bit:

```
        // get the current query string - if it's blank or matches the one we're trying to set to, do nothing. This prevents duplicate history entries
        var curQuery = window.location.search;
        if (curQuery == '')
          curQuery = '/';
        if (curQuery != qString)
          window.history.pushState('', '', qString);
```

**Comment by s3m1s0n1c** [comment]
I originally tested in V0.3.0 where the problem exists.

The issue doesn't exist in V0.1.8
https://www.rfspager.com/?q=Dungog? 
gets converted to 
https://www.rfspager.com/?q=Dungog%3F

Something to do with encoding ? into %3f

**Comment by s3m1s0n1c** [comment]
m4r5hyToday at 3:03 PM
2.0 beta is a bit unhappy about it
eventully it pops over tho

**Comment by davidmckenzie** [comment]
Got a fix coming

**Comment by davidmckenzie** [comment]
#279 with the fix - lets push out 0.3.1 once that's merged since it's security vuln

**Comment by s3m1s0n1c** [comment]
Confirmed #279 fixes the issue
